sylt-parser/src/expression.rs

```diff
@@ -340,41 +349,128 @@ fn unary<'t>(ctx: Context<'t>) -> ParseResult<'t, Expression> {
+
+    let lhs = Expression {
+        span: lhs.span,
+        kind: Duplicate(Box::new(lhs.clone())),
+    };
+
+    let (ctx, condition) = infix(ctx, &lhs)?;
+    let ctx = expect!(
+        ctx,
+        T::Else,
+        "Expected 'else' after short if-expression condition"
+    );
+    let (ctx, rhs) = parse_precedence(ctx, Prec::No)?;
+
+    let condition = Box::new(condition.clone());
+    let fail = Box::new(rhs);
+    Ok((
+        ctx,
+        Expression {
+            span,
+            kind: IfShort {
+                condition,
+                fail,
+            },
+        },
+    ))
+}
+
+
+fn if_expression<'t>(ctx: Context<'t>, lhs: &Expression) -> ParseResult<'t, Expression> {
+    let span = ctx.span();
+    let ctx = expect!(ctx, T::If, "Expected 'if' at start of if-expression");
+
+    use ExpressionKind::*;
+    let (ctx, condition) = parse_precedence(ctx, Prec::No)?;
+
+    let ctx = expect!(
+        ctx,
+        T::Else,
+        "Expected 'else' after if-expression condition"
+    );
+    let (ctx, rhs) = parse_precedence(ctx, Prec::No)?;
+    let condition = Box::new(condition.clone());
+    let pass = Box::new(lhs.clone());
+    let fail = Box::new(rhs);
+    Ok((
+        ctx,
+        Expression {
+            span,
+            kind: IfExpression {
+                condition,
+                pass,
+                fail,
+            },
+        },
+    ))
+}
+
+fn arrow_call<'t>(ctx: Context<'t>, lhs: &Expression) -> ParseResult<'t, Expression> {
+    let ctx = expect!(ctx, T::Arrow, "Expected '->' in arrow function call");
+    let (ctx, rhs) = expression(ctx)?;
+
+    use ExpressionKind::*;
+    use AssignableKind::{Call, ArrowCall};
+
+    fn prepend_expresion<'t>(ctx: Context<'t>, lhs: Expression, rhs: Expression) -> ParseResult<'t, Expression> {
+        let span = ctx.span();
+        let kind = match rhs.kind {
+            Get(Assignable {
+                kind: Call(callee, args),
+                ..
+            }) => Get(Assignable {
+                kind: ArrowCall(Box::new(lhs), callee, args),
+                span: rhs.span,
+            }),
+
+            Get(Assignable {
+                kind: ArrowCall(pre, callee, args),
+                ..
+            }) => {
+                let (_, pre) = prepend_expresion(ctx, lhs, *pre)?;
+                Get(Assignable {
+                    kind: ArrowCall(Box::new(pre), callee, args),
+                    span: rhs.span,
+                })
+            }
+
+            _ => { raise_syntax_error!(ctx, "Expected a call-expression after '->'"); }
+        };
+        Ok((ctx, Expression { span, kind }))
+    }
+
+    prepend_expresion(ctx, lhs.clone(), rhs)
+}
+
 /// Parse an expression starting from an infix operator. Called by `parse_precedence`.
 fn infix<'t>(ctx: Context<'t>, lhs: &Expression) -> ParseResult<'t, Expression> {
     use ExpressionKind::*;
 
+    match (ctx.token(), precedence(ctx.skip(1).token())) {
```

Comment:
Maybe a small comment here as to why we call precedence to check whether it's a normal if expression or a short if expression.